File: k-data-types/cpp/data_types.cpp
#include <iostream>
#include <string>
using namespace std;

int main(){

    // <<< primary/in-built data types >>>

    // int
    int no_of_space_crew = 7;
    // float
    float total_fuel = 9550.7;
    // double
    double miles_per_second = 2.339999999;
    // bool
    bool is_hydrogen_liquid = true;
    // void : absent of data type (often used with function)
    // char
    char is_water_supply_active = 'y';
    // wchar_t
    wchar_t is_main_engine_start = 'n';

    // <<< Derived data types >>>

    // function 
    // array
    string  mission_path[4] = {"moon","explore universe","mars","venus"};
    cout << mission_path[2] << endl;
    // string 
    string txt = "Welcome to Space";
    // pointer : get the memory address of a variable by using the `&` operator
    string longest_path_mission = "Venus";
    cout << longest_path_mission << '\n';
    cout << &longest_path_mission << endl; // Output the memory_ address
    // reference : store the address of another variable 
    string target_mission_tonight = "Moon";
    string &space_ship = target_mission_tonight;
    cout << target_mission_tonight << '\n';
    cout << space_ship << endl;

    // <<< User-defiend data types >>>

    // class
    // union
    // enum
    // struct
    // typedef 
    
    return 0;
}

File: k-data-types/python/data_types.py
def main():

    '''primitive data types'''
    # int
    no_of_space_crew = 7
    # float
    total_fuel = 2550.7
    # bool
    is_hydrogen_liquid = True
    # character
    is_water_supply_active = 'y'
    # str
    target_mission_tonight = "Moon"
    # complex
    a = 4 + 5j
    # None
    get_moon_temperature = None
    
    '''non-primitive data types'''
    # list 
    mission_path = ["moon","explore_universe","mars","venus"]
    # tuple
    capture_speed_per_hours = (1200,780,2599,1200,1250,1873,2500)
    # dict
    space_ship = {
        "no_of_crew" : 7,
        "color" : "silver",
        "path" : ["moon","venus","mars"]
    }
    # set
    space_ship_checking = {"door","engine","fuel","support_tank"}
    return
main()